### PR TITLE
[REF] pos*: simplify the NumberPopup

### PR DESCRIPTION
**Pull request by vlst-odoo**
In the number popup, in desktop mode we show the pos numpad,
while in mobile mode we show a regular input element.

This was done under the assumption that it would
be best if the user used their own's device native input method,
but this turned out to not be the case. In practice, incorrect
handling of this input proved to create many issues
( such as the one in task 3741406 ).

We now modify the number popup such that it always
displays the pos numpad and no longer relies on the input element.

Task 3743762
   
[IMP] point_of_sale: mobile dialogs at the bottom

In this commit we make the dialogs ( popups ) show up at the bottom of
the screen for small displays. This improves accesibility, as the dialog
will be closer to the users reach.

Before:
image

Now: 
image

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by JulienVR** [comment]
Hello @vlst-odoo, not sure whether this is  still a WIP, but a commit message would be nice :)

File: addons/point_of_sale/static/src/app/generic_components/numpad/numpad.js [hunk 1]
```diff
@@ -3,31 +3,64 @@
...
+                text: { type: String, optional: true },
+                class: { type: String, optional: true },
+                disabled: { type: Boolean, optional: true },
+            },
+        },
+        Number,
+        String,
+    ],
+};
+export function getButtons(env, rigthColumn) {
```
Comment (caburj) [inline addons/point_of_sale/static/src/app/generic_components/numpad/numpad.js:22]:
```suggestion
export function getButtons(env, rightColumn) {
```
misspelling

Comment (vlst-odoo) [reply addons/point_of_sale/static/src/app/generic_components/numpad/numpad.js:22]: True! my bad. It's fixed now

**Comment by caburj** [comment]
robodoo rebase-ff

**Comment by vlst-odoo** [comment]
robodoo retry

**Comment by caburj** [comment]
robodoo rebase-ff r+

**Pull request closed without merging**